scipy/spatial/_geometric_slerp.py

```diff
@@ -0,0 +1,206 @@
+from __future__ import division, print_function, absolute_import
+
+__all__ = ['geometric_slerp']
+
+import math
+import numpy as np
+from . import _slerp
+from scipy.spatial.distance import euclidean
+
+def geometric_slerp(start,
```

Comment:
Don't you think it can be useful to provide a class implementation, similar to, say, `PchipInterpolator` and `Slerp`?

The current function does interpolation along a single segment. Both can be provided.

Reply:
I'm not clear on the advantage of an object-oriented approach for this simple function? What is the use case, exactly?

Reply:
My suggestion was about making SLERP across many segments like quaternion SLERP [here](http://scipy.github.io/devdocs/generated/scipy.spatial.transform.Slerp.html#scipy.spatial.transform.Slerp).

This should reuse the function you implemented here and can be done separately. So let's don't consider it here.